### PR TITLE
fix: fix serialization issues (#438)

### LINKED ISSUE #436: RouteScopedContext.NavigationData is not serializable
Instances of that class are stored as UI attributes, so the class should implement the `Serializable` interface, to prevent exceptions in environments that rely on HTTP session replication.

### LINKED ISSUE #437: Serialization issue with UIEventListener
`CdiVaadinServletService` installs an instance of `UIEventListener` to listen to UI events.
However, the listener holds a reference to `VaadinService` through a `CdiVaadinServiceDelegate` instance.

`VaadinService` is not supposed to be stored in the HTTP session, and it also contains a reference to a `ClassLoader` instance, that is not serializable.

Either the `UIEventListener` should declare its `CdiVaadinServiceDelegate` field as transient, or the `CdiVaadinServiceDelegate` should mark its `VaadinService` reference as transient.

It should be checked if/how the `VaadinService` instance should be restored upon deserialization.

### PR DESCRIPTION
**Pull request by mcollovati**
Fixes #436
Fixes #437

### CODE REVIEW

**Review by czp13 — APPROVED** [review]

File: vaadin-cdi/src/main/java/com/vaadin/cdi/CdiVaadinServletService.java [hunk 1]
```diff
@@ -146,21 +161,17 @@ public CdiVaadinServlet getServlet() {
      */
     public static class CdiVaadinServiceDelegate implements Serializable {
 
-        private final VaadinService vaadinService;
-
         private transient BeanManager beanManager;
 
         private final UIEventListener uiEventListener;
 
         public CdiVaadinServiceDelegate(VaadinService vaadinService,
```
Comment (czp13) [inline vaadin-cdi/src/main/java/com/vaadin/cdi/CdiVaadinServletService.java:168]: Could we remove from here as well the `vaadinService`, if I am correct it is unused now.

Comment (czp13) [reply vaadin-cdi/src/main/java/com/vaadin/cdi/CdiVaadinServletService.java:168]: In that case probably from this line as well: 
`this.delegate = new CdiVaadinServiceDelegate(this, beanManager);`

Comment (mcollovati) [reply vaadin-cdi/src/main/java/com/vaadin/cdi/CdiVaadinServletService.java]: done

Comment (czp13) [reply vaadin-cdi/src/main/java/com/vaadin/cdi/CdiVaadinServletService.java]: Thank you vm! 🙇

**Review by czp13 — APPROVED** [review]
Checked again, more deeply, code is good, tests are covering changes! Feel safe to merge.

**Pull request merged**